Plugin/OgoneBatchGatewayPlugin.php

```diff
@@ -0,0 +1,452 @@
+                'USERID' => $this->token->getApiUser(),
+            );
+
+            $this->logger->debug('Checking transaction status with Ogone with params {params}', array('params' => $params));
+            $xmlResponse = $this->sendApiRequest(array(), $this->getDirectQueryUrl().'?'.http_build_query($params), 'GET');
+
+            $response = new DirectResponse($xmlResponse);
+            $this->logger->debug('response status is {status}', array('status' => $response->getStatus()));
+
+            if (!$response->hasError() && $response->isIncomplete()) {
+                $transaction->setReferenceNumber($response->getPaymentId());
+                $this->logger->debug('Authorization transaction send', array('res' => $response));
+                throw new PaymentPendingException(sprintf('Authorization is still pending, status: %s.', $response->getStatus()));
+            } else {
+                $this->logger->debug('Authorization is not successful');
+                $this->handleUnsuccessfulResponse($response, $transaction);
+            }
+        } else {
+            $this->logger->debug('feedback response set.');
+            $response = $this->feedbackResponse;
+        }
+
+        if ($response->isApproving()) {
+            $this->logger->debug('response {res} is still approving', array('res' => $response));
+            throw new PaymentPendingException(sprintf('Payment is still approving, status: %s.', $response->getStatus()));
+        }
+
+        if (!$response->isApproved()) {
+            $this->logger->debug('response {res} is not approved', array('res' => $response));
+            $ex = new FinancialException(sprintf('Payment status "%s" is not valid for approvment', $response->getStatus()));
+            $ex->setFinancialTransaction($transaction);
+            $transaction->setResponseCode($response->getErrorCode());
+            $transaction->setReasonCode($response->getStatus());
+
+            throw $ex;
+        }
+
+        $transaction->setProcessedAmount($response->getAmount());
+        $transaction->setResponseCode(PluginInterface::RESPONSE_CODE_SUCCESS);
+        $transaction->setReasonCode(PluginInterface::REASON_CODE_SUCCESS);
+    }
+
+    /**
+     * This method executes a deposit transaction (aka capture transaction).
+     *
+     * This method requires that the Payment has already been approved in
+     * a prior transaction.
+     *
+     * A typical use case are Credit Card payments.
+     *
+     * @param FinancialTransactionInterface $transaction The transaction
+     * @param boolean                       $retry       Retry
+     *
+     * @return mixed
+     *
+     * @throws ActionRequiredException If the transaction's state is NEW
+     * @throws FinancialException      If payment is not approved
+     * @throws PaymentPendingException If payment is still approving
+     */
+    public function deposit(FinancialTransactionInterface $transaction, $retry)
```

Comment:
please separate this method in readable & thins methods
(same for approve and reverseDeposit)